src/middlewared/middlewared/plugins/pool_/dataset.py

```diff
@@ -302,6 +302,9 @@ async def __common_validation(self, verrors, schema, data, mode, parent=None, cu
                 if acltype in ['POSIX', 'OFF'] and to_check.get('aclmode', 'DISCARD') != 'DISCARD':
                     verrors.add(f'{schema}.aclmode', 'Must be set to DISCARD when acltype is POSIX or OFF')
 
+                elif acltype == 'NFSV4' and to_check.get('aclmode') == 'DISCARD':
```

Comment:
Can `aclmode` be not specified ? And should we have a default like we do in 302 ?